### PR TITLE
0.4 roadmap

### PR DESCRIPTION
**Pull request by sofroniewn**
# Description
This PR contains a roadmap for our 0.4 series of releases. It also has a look back for our 0.3 series. Right now this is inline with the old 0.3 roadmap, but we could put it at the top of the 0.4 roadmap or in its own document.

This is still very early, but feedback always welcome!

### CODE REVIEW

**Comment by tlambert03** [comment]
I'd like to see a bullet point about refactoring/cleaning existing code.  I don't think any sort of "feature freeze" is in order any time soon, but I do think we should keep revisiting what has been done and refining/clarifying/documenting it... so always good to have that goal explicitly mentioned.

**Comment by sofroniewn** [comment]
> I'd like to see a bullet point about refactoring/cleaning existing code.

Good point - we can see how specific we want to get, but we've got a lot in progress or planned, evented dataclass, color manager, slicing off the layer, asycn clean up, console pullout, more qt / model separation (own model repo?), among others.

Is this the sort of stuff you had in mind @tlambert03? Any of those you want to make sure we mention? @jni already asked for async clean up to go in its corresponding section.

**Comment by tlambert03** [comment]
Yep that kind of stuff!

**Comment by justinelarsen** [comment]
Overall, I think the 0.4 roadmap looks great! I like the inline comments for the look back on the 0.3 series. It helps to provide context and track progress. I also agree with @tlambert03 on calling out the refactoring/cleanup explicitly so it's clear and we make progress on it.

**Comment by sofroniewn** [comment]
Just noting that an in place updatable version of the bundled app was identified as a high priority need

**Comment by sofroniewn** [comment]
Ok @jni this has been updated so 0.3 roadmap lookback is now its own doc and 0.3 roadmap is unchanged

File: docs/source/developers/ROADMAP_0_3_lookback.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
+# Roadmap
+
+## For 0.3.* series of releases - April 2020
+
```
Comment (jni) [inline docs/source/developers/ROADMAP_0_3_lookback.md]: I would change the title/subtitle here, e.g.

```suggestion
# Roadmap retrospective

Periodically, the napari team produces roadmaps to clarify priorities in the coming months of development. When a new roadmap is introduced, we will retrospectively examine how well actual development tracked with the appropriate roadmap.

Below, we examine our 0.3.x roadmap (reproduced in its entirety) and compare it to what we achieved in the corresponding 6 months (in inline quote blocks).

## For 0.3.* series of releases - April 2020-Nov 2020

```

Comment (jni) [reply docs/source/developers/ROADMAP_0_3_lookback.md]: I'd also consider moving the initial quote block to its own paragraph above the `## For 0.3`.

I also feel uncomfortable saying stuff like "we've reduced the number of bugs" without numbers. We can find this out using GH search, finding number of issues called "bug" that were closed in that period, and of those, how many were actually there when we created the roadmap — ie how much effort was actually spent on clearing up existing bugs vs creating and fixing new ones. I can try to do this after this review.

File: docs/source/developers/ROADMAP_0_3_lookback.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+- **Better support for viewing big datasets**. Currently, napari is fast when viewing on-disk datasets that can be naturally sliced along one axis (e.g. a time series) *and where loading one slice is fast*. However, when the loading is slow, the napari UI itself becomes slow, sometimes to the point of being unusable. We aim to improve this by making views and interactions non-blocking ([#845](https://github.com/napari/napari/issues/845)), and improving caching ([#718](https://github.com/napari/napari/issues/718)). We will also ensure that napari can be used `headless` without the GUI.
+
+    > We've made good progress towards `asynchronous` loading of data in our [async](https://github.com/napari/napari/labels/async) work, but we still have much more to do. The async functionality can be enabled right now in an opt in fashion but is still considered experimental as we develop it further. We're also working on an octree  which will enable better multiscale rendering, see this [rendering explanation note](https://napari.org/docs/dev/explanations/rendering.html). We will continue this work during the 0.4 series of releases.
```
Comment (jni) [inline docs/source/developers/ROADMAP_0_3_lookback.md]: octree is also now available.

File: docs/source/developers/ROADMAP_0_3_lookback.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+
+- **Improving the performance of operations on in-memory data**. Even when data is loaded in memory, some operations, such as label and shape painting, slicing along large numbers of points, or adjusting contrast and gamma, can be slow. We will continue developing our [benchmark suite](https://github.com/napari/napari/blob/master/docs/developers/BENCHMARKS.md) and work to integrate it into our development process. See the [`performance` label](https://github.com/napari/napari/labels/performance) for a current list of issues related to performance.
+
+    > We've improved some perfomance issues, such as adjusting gamma and contrast limits, which now happen on the shader, but still have more to go. Painting in labels remains slow for large datasets and still needs to be improved. We added an [explantory note on perfomance](https://napari.org/docs/dev/explanations/performance.html) to detail how we're thinking about perfomance in napari and ensuring it stays high. We've also added some [perfomance monitoring tooling](https://napari.org/docs/dev/events/perfmon.html) that will help us during this process. We will continue this work during the 0.4 series of releases.
```
Comment (jni) [inline docs/source/developers/ROADMAP_0_3_lookback.md]: Note here that we successfully used perfmon to dramatically improve the performance of the shapes layer

File: docs/source/developers/ROADMAP_0_3_lookback.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+- Add a unified **world coordinate system**. Scientists need to measure data that comes from a real space with physical dimensions. Currently, napari has no concept of the space in which data lives: everything is unitless. Further, it is unclear at various parts in the UI whether a coordinate has been transformed. And finally, some data are acquired with distortions, such as skew in data collected on stage-scanning lightsheet microscopes, and napari should be able to account for those distortions by chaining together transforms - including affine and ultimately deformable transforms. We are tracking progress in this area in the [World Coordinates project board](https://github.com/napari/napari/projects/10).
+
+    > We have completed our world coordinate system, including support for a affine transforms, a scale bar, and an axes rotation visual. We have more work to do to add support for physical units which will come in the 0.4 series of releases, non-orthogonal slicing, and non-rigid transforms, which will be saved for future roadmaps.
```
Comment (jni) [inline docs/source/developers/ROADMAP_0_3_lookback.md]: a affine -> affine

File: docs/source/developers/ROADMAP_0_3_lookback.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+- Ensure **easy installation with a success rate close to 100%**. We can still struggle with installation of Qt in some cases, and have had problems with dependency conflicts with other Python packages. We have recently added a [conda-forge package](https://github.com/conda-forge/napari-feedstock) and are working towards distributing bundled applications in [#496](https://github.com/napari/napari/pull/496). See the [`installation` label](https://github.com/napari/napari/labels/installation) for a current list of issues related to installation.
+
+    > We have made good progress on improving the success rate of our installation process, and now encounter less installation related new issues.
```
Comment (jni) [inline docs/source/developers/ROADMAP_0_3_lookback.md]: This undersells our progress, I think!

- installation in fresh environments is often successful, especially using conda-forge
- the bundled app exists and works for a large proportion of users
- *but* many platform specific bugs still remain. Windows is often a problem, partly due to lack of representation in the core team, and macOS Big Sur has raised a whole new set of issues, though again, this works with conda-forge and python 3.9.0 🎉

File: docs/source/developers/ROADMAP_0_3_lookback.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+    > We've reduced the number of open [`bug`](https://github.com/napari/napari/labels/bug) labeled issues by more than 20% in the past two months from > 80 to < 60 and added some more screenshot and GUI tests, but test coverage of the GUI can still be improved.
```
Comment (jni) [inline docs/source/developers/ROADMAP_0_3_lookback.md]: Ah great stuff. I'd suggest mentioning this in the intro *or* writing (see below)

File: docs/source/developers/ROADMAP_0_3_lookback.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+
+- **Improve support for annotation** of points [#858](https://github.com/napari/napari/issues/858), shapes [#177](https://github.com/napari/napari/issues/177), labels, and images [#209](https://github.com/napari/napari/issues/209) including text rendering [#600](https://github.com/napari/napari/pull/600).
+
+    > We've now added support to label points with text and somewhat improved color handling for our points, shapes, and labels layers, but have more work to do to standardize those improvements and reduce code duplication.
```
Comment (jni) [inline docs/source/developers/ROADMAP_0_3_lookback.md]: Again, this feels like we did very little, but actually all of that work is great and works really well! I would rephrase as "we have added programmatic support for text and other properties in points, shapes, and labels, allowing users to effectively visualize many properties in their data. However, support for editing these properties from the UI remains limited or non-existent, so we will aim to improve this in the next roadmap."

File: docs/source/developers/ROADMAP_0_3_lookback.md [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+
+    > We have conducted a product heuristics analysis to [identify design and usability issues](https://github.com/napari/product-heuristics-2020), and will now be working during the 0.4 series of releases to implement them.
+
+## Make a downloadable application with basic plugin management and persistent settings
+
+- Distribute **a bundled application for each major OS** [#496](https://github.com/napari/napari/pull/496) to allow scientists to use napari without requiring a Python development environment, which can be difficult to install and maintain. The bundle should contain the necessary machinery to add plugins [#1074](https://github.com/napari/napari/issues/1074).
+
+    > We have succeeded in providing a bundled app that you can install plugins into. We have more work to do around plugin management that will continue in our 0.4 series of releases.
```
Comment (jni) [inline docs/source/developers/ROADMAP_0_3_lookback.md]: I don't know how to improve this but it sounds so matter of fact when it is indeed so awesome! 😂

File: docs/source/developers/ROADMAP_0_3_lookback.md [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+
+- Support for reader plugins [#937](https://github.com/napari/napari/pull/937) and writer plugins [#1068](https://github.com/napari/napari/issues/1068) to allow **viewing of domain-specific data and saving of annotations**. For more details see the [`plugins` label](https://github.com/napari/napari/labels/plugins) on our repository.
+
+    > We have a fairly stable specification for adding both [reader and writer plugins](https://napari.org/docs/dev/plugins/hook_specifications.html) that is being used in the community.
```
Comment (jni) [inline docs/source/developers/ROADMAP_0_3_lookback.md]: remove fairly?

**Review by jni — COMMENTED** [review]
Just added some notes on the retrospective, submitting now but will add further notes for the 0.4 roadmap

File: docs/source/developers/ROADMAP_0_3_lookback.md [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+
+- Support for generating animations [#780](https://github.com/napari/napari/pull/780). This feature will be supported after we have the ability to serialize the viewer state [#851](https://github.com/napari/napari/pull/851).
+
+- Draggable and resizable layers [#299](https://github.com/napari/napari/issues/299) and [#989](https://github.com/napari/napari/pull/989). This feature will be supported after we have added support for world coordinates [project board 10](https://github.com/napari/napari/projects/10) and rotations to our transform model.
+
+- Linked 1D plots such as histograms, timeseries, or z-profiles [#823](https://github.com/napari/napari/pull/823) and [#675](https://github.com/napari/napari/pull/675).
+
+- Support for using napari with remote computation (i.e. a remote jupyter notebook [#495](https://github.com/napari/napari/issues/495)).
+
+- Functional or interactive plugins that allow for analysis of data or add elements to the GUI.
```
Comment (jni) [inline docs/source/developers/ROADMAP_0_3_lookback.md]: Worth writing here that we actually have experimental support for this right now!

File: docs/source/developers/ROADMAP_0_4.md [hunk 11]
```diff
@@ -0,0 +1,90 @@
+# Roadmap
+
+## For 0.4.* series of releases - November 2020
```
Comment (jni) [inline docs/source/developers/ROADMAP_0_4.md:3]: January 2021?

File: docs/source/developers/ROADMAP_0_4.md [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+
+- Make the **data viewing and annotation** capabilities **bug-free, fast, and delightful to use**.
+
+- Support **functional and interactive** plugins that allow for image analysis.
+
+- Add accessible documentation, tutorials, and demos.
+
+You can read more about how this roadmap builds on and continues the work in our `0.3` series of releases in our [0.3 lookback section](#lookback-at-the-0.3-roadmap). We're continuing to prioritize the robustness and polish of the core viewer to ensure that plugin development happens against a solid foundation, and because we believe that looking at and annotating your data should be a bug free and delighful experience, regardless of how large it is. We want to build on the success of our reader / writer plugins to provide the ability to add functional and interactive plugins and further empower a lot of the great work being built ontop of napari.
```
Comment (jni) [inline docs/source/developers/ROADMAP_0_4.md]: The link probably needs updating

File: docs/source/developers/ROADMAP_0_4.md [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+
+## Make the data viewing and annotation capabilities bug-free, fast, and delightful to use
+
+- ...
+
+- **Better support for viewing big datasets**. Currently, napari is fast when viewing on-disk datasets that can be naturally sliced along one axis (e.g. a time series) *and where loading one slice is fast*. However, when the loading is slow, the napari UI itself becomes slow, sometimes to the point of being unusable. We now have experimental support for asyncronous rendering for our images, and are in the process of adding support for an octree to better support multiscale tiled data for both 2D and 3D rendering.
```
Comment (jni) [inline docs/source/developers/ROADMAP_0_4.md]: Both of these are now implemented but as opt-in, only programmatically. I would outline two priorities here:
- add persistent settings and use those to allow anyone to use these features, even in the bundled app
- bring the features out of experimental and into mainline napari

File: docs/source/developers/ROADMAP_0_3_lookback.md (hunk 1 above)
Comment (sofroniewn) [reply docs/source/developers/ROADMAP_0_3_lookback.md]: good points on the organization

on the bug side, we have > 80 when i started tracking over the summer (~July) and then brought that down to < 60 in November. We're now back at 64. Agreed though those comments are best supported with data, which is easy to produce. We can tone it down for the retro, but would be nice to do those sorts of analyses at some point and include figures!

File: docs/source/developers/ROADMAP_0_3_lookback.md (hunk 2 above)
Comment (sofroniewn) [reply docs/source/developers/ROADMAP_0_3_lookback.md]: will update

File: docs/source/developers/ROADMAP_0_3_lookback.md (hunk 3 above)
Comment (sofroniewn) [reply docs/source/developers/ROADMAP_0_3_lookback.md]: will add

File: docs/source/developers/ROADMAP_0_3_lookback.md (hunk 6 above)
Comment (sofroniewn) [reply docs/source/developers/ROADMAP_0_3_lookback.md]: ah it's down here! i forgot i had put that in, great!

File: docs/source/developers/ROADMAP_0_3_lookback.md (hunk 9 above)
Comment (sofroniewn) [reply docs/source/developers/ROADMAP_0_3_lookback.md]: yes that's fair ;-)

File: docs/source/developers/ROADMAP_0_4.md [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+
+- Add **physical coordinates**. We now have a world coordinate system and transforms that can move between data coordinates, world coordinates, and the canvas where things are rendered; however, we still don't have a concept of phyiscal units. See [#1701](https://github.com/napari/napari/issues/1701) for additional discussion.
+
+- Add **linked multi-canvas support** ([#760](https://github.com/napari/napari/issues/760)) to allow orthogonal views, or linked views of two datasets, for example to select keypoints for image alignment, or simultaneous 2D slices with 3D rendering.
+
+- Add **layer groups** [#970](https://github.com/napari/napari/issues/970), which allow operating on many layers simultaneously making the viewer easier to use for multispectral or multimodal data, or, in the context of multiple canvases, where one wants to assign different groups to different canvases.
+
+- Complete **serialization of the viewer** [#851](https://github.com/napari/napari/pull/851) to enable sharing the entire viewer state, see [#1875](https://github.com/napari/napari/issues/1875) too.
```
Comment (jni) [inline docs/source/developers/ROADMAP_0_4.md]: I don't love this as a priority on its own — it'd be good to mention what this enables in terms of usage and features, e.g. remote viewer, sharing of viewers between different users, deep linking, animation interpolating all attributes of the viewer...

File: docs/source/developers/ROADMAP_0_4.md [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+
+- Support for generating **animations** [#780](https://github.com/napari/napari/pull/780).
```
Comment (jni) [inline docs/source/developers/ROADMAP_0_4.md]: maybe link to your repo here? Or remove altogether — maybe point to your repo in the retrospective?

File: docs/source/developers/ROADMAP_0_4.md [hunk 16]
```diff
@@ -0,0 +1,90 @@
...
+
+- Improved **error handling and bug reporting**, see [#1090](https://github.com/napari/napari/issues/1090) for details.
+
+- Support for persistent settings [#1183](https://github.com/napari/napari/pull/1183) to allow **saving of preferences** between launches of the viewer.
```
Comment (jni) [inline docs/source/developers/ROADMAP_0_4.md]: This should be bumped up, and we should mention that it is critical to give users early access to experimental features.

File: docs/source/developers/ROADMAP_0_4.md [hunk 17]
```diff
@@ -0,0 +1,90 @@
...
+
+- Improve the **user interface and design** of the viewer to make it easier to use. We have conducted a product heuristics analysis to [identify design and usability issues](https://github.com/napari/product-heuristics-2020), and will now be working during the 0.4 series of releases to implement them. Also see the [`design` label](https://github.com/napari/napari/labels/design) for more information.
+
+## Support **functional and interactive** plugins that allow for image analysis.
+
+- Support for functional plugins that take in layers or layer data and operate on them to return new layers or layer data.
```
Comment (jni) [inline docs/source/developers/ROADMAP_0_4.md]: Already done. Maybe, strengthen support? It would be nice actually to tie it in with the community, e.g. "0.4.3 brought experimental support for functional plugins. During the 0.4.x series, we aim to work with our community to strengthen that support, identify gaps in functionality, and fill those gaps, either with new plugin types or by modifying the existing plugins architecture to suit developer and user needs."

File: docs/source/developers/ROADMAP_0_4.md [hunk 18]
```diff
@@ -0,0 +1,90 @@
...
+
+Our [0.3 roadmap](ROADMAP_0_3.md) had the following main prioities:
+
+- Make the **data viewing and annotation** capabilities **bug-free and fast**.
+
+- Make a **downloadable application** with reader / writer plugin management.
+
+- Make accessible documentation, tutorials, and demos.
+
+Overall we made good progress towards these priorites. We now provide cross-platform support for downloadable app that you can install reader and writer plugins into. We added support for world coordinates, we reduced the number of bugs, and we've begun begun to support asynchronous rendering and improve performance. We also had some highlights that weren't on our roadmap including contributions from the community like the Tracks layer. We've added more documentation and tutorials, but a redesign of our website is still in progress. For a detailed lookback at the work that was done during the `0.3` series of releases you can look at comments under the individual bullet items in our [0.3 roadmap](ROADMAP_0_3.md) that we have added as we made this roadmap.
```
Comment (jni) [inline docs/source/developers/ROADMAP_0_4.md]: Delete?

File: docs/source/developers/ROADMAP_0_4.md [hunk 19]
```diff
@@ -0,0 +1,90 @@
...
+
+The napari roadmap captures current development priorities within the project and aims to guide for napari core developers, to encourage and inspire contributors, and to provide insights to external developers who are interested in building for the napari ecosystem. For more details on what this document is and is not, see the [about this document section](#about-this-document).
+
+The [mission](MISSION_AND_VALUES.md#our-mission) of napari is to be a foundational multi-dimensional image viewer for Python and provide graphical user interface (GUI) access to image analysis tools from the Scientific Python ecosystem for scientists across domains and levels of coding experience. To work towards this mission, we have set the following three high-level priorities to guide us over the upcoming months:
+
+- Make the **data viewing and annotation** capabilities **bug-free, fast, and delightful to use**.
+
+- Support **functional and interactive** plugins that allow for image analysis.
+
+- Add accessible documentation, tutorials, and demos.
```
Comment (jni) [inline docs/source/developers/ROADMAP_0_4.md]:
```suggestion
- Add accessible **documentation**, tutorials, and demos.
```

File: docs/source/developers/ROADMAP_0_4.md [hunk 20]
```diff
@@ -0,0 +1,90 @@
...
+
+## For 0.4.* series of releases - November 2020
+
+The napari roadmap captures current development priorities within the project and aims to guide for napari core developers, to encourage and inspire contributors, and to provide insights to external developers who are interested in building for the napari ecosystem. For more details on what this document is and is not, see the [about this document section](#about-this-document).
+
+The [mission](MISSION_AND_VALUES.md#our-mission) of napari is to be a foundational multi-dimensional image viewer for Python and provide graphical user interface (GUI) access to image analysis tools from the Scientific Python ecosystem for scientists across domains and levels of coding experience. To work towards this mission, we have set the following three high-level priorities to guide us over the upcoming months:
+
+- Make the **data viewing and annotation** capabilities **bug-free, fast, and delightful to use**.
+
+- Support **functional and interactive** plugins that allow for image analysis.
```
Comment (jni) [inline docs/source/developers/ROADMAP_0_4.md]:
```suggestion
- Support **functional and interactive** plugins that enable complete image analysis workflows.
```

**Review by jni — APPROVED** [review]
Ok I've added a review for the 0.4 roadmap itself. One high level comment I would add is that each section should have a little introductory paragraph for it before launching into the dot-points. Happy to write those myself.

File: docs/source/developers/ROADMAP_0_3_lookback.md (hunk 5 above)
Comment (sofroniewn) [reply docs/source/developers/ROADMAP_0_3_lookback.md]: done

File: docs/source/developers/ROADMAP_0_3_lookback.md (hunk 6 above)
Comment (sofroniewn) [reply docs/source/developers/ROADMAP_0_3_lookback.md]: done

File: docs/source/developers/ROADMAP_0_3_lookback.md (hunk 10 above)
Comment (sofroniewn) [reply docs/source/developers/ROADMAP_0_3_lookback.md]: added line, don't want to say too much as a feel like it belongs more in our 0.4 section

Comment (sofroniewn) [reply docs/source/developers/ROADMAP_0_3_lookback.md]: wasn't really an 0.3 accomplishment, i might actually not include. it will be highlighted elsewhere

File: docs/source/developers/ROADMAP_0_3_lookback.md (hunk 7 above)
Comment (sofroniewn) [reply docs/source/developers/ROADMAP_0_3_lookback.md]: done

File: docs/source/developers/ROADMAP_0_3_lookback.md (hunk 8 above)
Comment (sofroniewn) [reply docs/source/developers/ROADMAP_0_3_lookback.md]: ok more added

File: docs/source/developers/ROADMAP_0_4.md (hunk 11 above)
Comment (sofroniewn) [reply docs/source/developers/ROADMAP_0_4.md:3]: well strictly speaking the first 0.4 release was back then and the doc started in november. i think it's ok. we probably shouldn't let it sit as long next time, but i'd probably leave as is now

File: docs/source/developers/ROADMAP_0_4.md (hunk 18 above)
Comment (sofroniewn) [reply docs/source/developers/ROADMAP_0_4.md]: yeah, we can cut section

File: docs/source/developers/ROADMAP_0_4.md (hunk 12 above)
Comment (sofroniewn) [reply docs/source/developers/ROADMAP_0_4.md]: updated

File: docs/source/developers/ROADMAP_0_4.md (hunk 13 above)
Comment (sofroniewn) [reply docs/source/developers/ROADMAP_0_4.md]: can you take this one

File: docs/source/developers/ROADMAP_0_4.md (hunk 14 above)
Comment (sofroniewn) [reply docs/source/developers/ROADMAP_0_4.md]: hmm that's fair. I do want to capture this feedback too https://github.com/napari/napari/pull/1906#issuecomment-730779928

Comment (sofroniewn) [reply docs/source/developers/ROADMAP_0_4.md]: open to ideas/ suggestions, maybe you can take a pass?

File: docs/source/developers/ROADMAP_0_4.md (hunk 16 above)
Comment (sofroniewn) [reply docs/source/developers/ROADMAP_0_4.md]: yes agreed, can you take

File: docs/source/developers/ROADMAP_0_4.md (hunk 17 above)
Comment (sofroniewn) [reply docs/source/developers/ROADMAP_0_4.md]: yes very open to that. i'd like some of the perspective of the doc to still be from everything in 0.4, but should be up to date with where we are now. can you take this one

File: docs/source/developers/ROADMAP_0_4.md (hunk 18 above)
Comment (sofroniewn) [reply docs/source/developers/ROADMAP_0_4.md]: done

File: docs/source/developers/ROADMAP_0_4.md (hunk 15 above)
Comment (sofroniewn) [reply docs/source/developers/ROADMAP_0_4.md]: ok, maybe we cut here and I add link to my repo in retro

Comment (sofroniewn) [reply docs/source/developers/ROADMAP_0_4.md]: done

File: docs/source/developers/ROADMAP_0_4.md (hunk 11 above)
Comment (jni) [reply docs/source/developers/ROADMAP_0_4.md:3]: eh, still feels like backdating a contract, but 🤷. Happy to leave it be.

**Comment by jni** [comment]
@tlambert03 I've written a paragraph about cleanup and refactor in 7cfde24c. Let me know if it captures the gist of what you envisioned it to say. 😊 (cc also @sofroniewn @justinelarsen)

I'm done making changes to the roadmap so further reviews are welcome! Hopefully not much more to be done before merging and cutting the release candidate! 🎉 CC @royerloic too.

**Comment by sofroniewn** [comment]
@jni your edits look great to me! I think it might be a little ambitious to get the asycn & hook_specs out of experimental before 0.5.0, I don't see that as a must, but it's ok to call out that's a goal.

If no one else has any additional comments I'll merge soon!

**Review by tlambert03 — APPROVED** [review]

**Comment by jni** [comment]
> it might be a little ambitious to get the asycn & hook_specs out of experimental before 0.5.0

later... "we are excited to announce napari 0.4.48!" 😜
